### PR TITLE
Feature/ppca regularized adversarial

### PR DESCRIPTION
**Pull request by austinTalbot7241993**

### CODE REVIEW

File: python/python/bystro/supervised_ppca/ppca_adversarial.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import numpy as np
+import numpy.linalg as la
+
+from bystro.supervised_ppca._base import BaseGaussianFactorModel
+from numpy.typing import NDArray
+
+
+class PPCAadversarial(BaseGaussianFactorModel):
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/ppca_adversarial.py:8]: Let's add a docstring explaining the variables being passed in

File: python/python/bystro/supervised_ppca/ppca_regularized.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    """
+    Analytic PPCA solution as described by Bishop
+
+    Parameters
+    ----------
+    n_components : int,default=2
+        The latent dimensionality
+    """
+
+    def __init__(self, n_components: int = 2, regularization_options=None):
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/ppca_regularized.py:23]: could we also document the regularization options?

**Review by akotlar — CHANGES_REQUESTED** [review]
Docstring improvements for PPCARegularized and PPCAadversarial

File: python/python/bystro/supervised_ppca/tests/test_ppca_regularized.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+def test_fit():
+    """Test the fitting process of the PPCARegularized model."""
+    # Create a synthetic dataset
+    rng = np.random.default_rng(2021)
+    X = rng.normal(size=(100, 5))  # 100 samples and 5 features
+
+    pca_model = PPCARegularized(
+        n_components=2, regularization_options={"method": "NonLinearShrinkage"}
+    )
+    pca_model.fit(X)
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/tests/test_ppca_regularized.py:22]: Something is going wrong here:

```python
In [3]: pca_model.fit(X)
/mnt/ssd1/bystro/python/python/bystro/supervised_ppca/ppca_regularized.py:75: RuntimeWarning: invalid value encountered in sqrt
  W = eigenvectors[:, :L] * np.sqrt(
Out[3]: <bystro.supervised_ppca.ppca_regularized.PPCARegularized at 0x7fb0cace1c90>
```

Comment (austinTalbot7241993) [reply python/python/bystro/supervised_ppca/tests/test_ppca_regularized.py:22]: Forgot to reorder the eigenvalues as well. Issue is now fixed.

**Review by akotlar — APPROVED** [review]
LGTM. Exciting addition! Thanks Austin.

**Pull request merged**